pkg/core/hubmanager.go

```diff
@@ -96,10 +97,10 @@ func NewHubManager(newHub hubClientCreator, stop <-chan struct{}) *HubManager {
 }
 
 // SetHubs ...
-func (hm *HubManager) SetHubs(hubURLs []string) {
+func (hm *HubManager) SetHubs(hubs map[string]*Host) {
 	newHubURLs := map[string]bool{}
-	for _, hubURL := range hubURLs {
-		newHubURLs[hubURL] = true
+	for _, hub := range hubs {
```

Comment:
You don't need to do this anymore.  The original code was converting an array into a map for a simple test for existence.  SetHubs now is receiving a map, so should be able to use the map that is passed in the same way.

Reply:
Fixed it